packages/core/src/core.ts

```diff
@@ -635,31 +654,35 @@ class ThresholdKey implements ITKey {
     importedKey?: BN;
     delete1OutOf1?: boolean;
   } = {}): Promise<InitializeNewKeyResult> {
-    if (!importedKey) {
-      const tmpPriv = generatePrivate();
+    let seed: Uint8Array = new Uint8Array();
+    if (this.keyType === KeyType.secp256k1) {
+      const tmpPriv = importedKey || generatePrivate(this.keyType);
       this._setKey(new BN(tmpPriv));
     } else {
-      this._setKey(new BN(importedKey));
+      seed = importedKey ? new Uint8Array(importedKey.toBuffer()) : nacl.randomBytes(32);
```

Comment:
This sometimes fails. The imported key is a BN, but the seed needs to be a 32 byte array. Depending on how BN is generated, it may only be 31 bytes.

Reply:
This would be a valid failure due to incorrect input then. No?

Reply:
The problem is that `BN.toBuffer` is not guaranteed to output 32 bytes but `nacl.sign.keyPair.fromSeed(seed);` on line 663 requires the seed to be 32 bytes. For example, consider the case where the 32-byte seed starts with a zero.

When I ran the tests locally, this was failing. In fact, the test (see below) was not even generating a 32 byte seed, but instead was generating a random scalar.

https://github.com/tkey/tkey/blob/15197375a8b792fa13b54066d6f0a3336690fdd0/packages/default/test/shared.js#L116

I think it's confusing that the imported key is a BN in the context of ed25519. An alternative would be to have the imported key be a byte array that either gets converted to BN (in case of secp256k1) or is consumed directly as a seed (in case of ed25519). What do you think?

Reply:
Seems to be resolved in https://github.com/tkey/tkey/pull/266/commits/15197375a8b792fa13b54066d6f0a3336690fdd0